### DISCUSSION TITLE
Delete state on cards (Office hours: Oct 13, 2020)

### DISCUSSION THREAD

**Discussion by pguptaDS** [twilio-labs/paste#843] [Office hours] [answered]
Would it be possible to have a red border on a card to communicate that card will be deleted once the page is saved? Below is what I am proposing:
![image](https://user-images.githubusercontent.com/64559618/95789127-cd5ac000-0c91-11eb-87ad-79e672cf81dc.png)

**Comment by kate-porter** [comment]
Hey @pguptaDS - could you give a high-level walkthrough of the steps taken to get the user into this state? I'm having trouble understanding exactly what this red border is communicating.

**Reply by pguptaDS** [reply]
@katieporter Yes absolutely. It might be helpful for you to see the interaction here. Click on "Delete Assignment" and go through the delete flow: https://twilio.invisionapp.com/share/4YZ2VNSAPFD#/screens/432056666_Edit_Assignment_-_3

What we learned during testing was that users did not realize that they need to click on Save after they hit "ok,got it" in the modal. Some users even read the text and still thought that their assignment was deleted. The main reason for this behavior was that all the users got visual feedback that the assignment was deleted because the card wasn't there on the screen afterwards. We still have guardrails after this as we show the user that there are pending changes if they navigate away from this page but we'd like to improve this experience by keeping the deleted card and visually indicating that the card will be deleted upon saving. 

Hope that helps!

**Reply by kate-porter** [reply]
Thank you, that context is super helpful. I needed to request access to the Invision prototype, but I've still got a much better grasp on the flow based on your explanation. Let me think on this a bit and get back to ya.

**Reply by pguptaDS** [reply]
@katieporter Awesome! Oh and I just updated the link so you shouldn't need to request access anymore. Please let me know if thats not the case.

**Reply by kate-porter** [reply]
My first thought is to compose an inline warning inside the card, rather than making the card border red, as the red border looks more like an error state and isn't communicating to the user why it's red.

SendGrid has a somewhat similar scenario in its Automation tool, where a user can edit an Automation but needs to save their changes before they go live, and they use an inline warning message, like so:

![image](https://user-images.githubusercontent.com/5374045/95889570-ca1e0d80-0d3f-11eb-8679-e117a7a08ec4.png)

We're working on a creating a standard pattern for a message like this, but in the meantime, we could help you compose something during office hours today if you think an inline message like this could work.

**Reply by SiTaggart** [reply]
Is the fact the delete action isn't actually a delete (you have to save the update to persist it) a technical constraint? The fact you trow up a confirmation modal and then don't actually follow through feels pretty broken to me. Like the operation failed, and hitting the save button is a retry.

Any possibility to complete the operation upon clicking the confirmation modal?

**Reply by serifluous** [reply]
^ In a similar vein, I have a feeling the command bar might be complicating things on this page. Does the user need to make a bunch of assignments _and then_ save, or can you save each role assignment for the user as they assign them (maybe pop a toast each time for extra confirmation), potentially removing the command bar?

Sidenote: Is there a chance that the "Make another assignment" button could get lost as more role assignments are added, and the button gets pushed further down the page?

**Comment by corinne-nagel** [comment, marked as answer]
Hi @pguptaDS , thanks for coming to office hours. I know we offered a lot of ideas, but I think we landed on one that was a composition of an alert without the background/close icon. Did you decide what direction to go? Let us know if you have any further questions on this or if we closed the loop yesterday!

**Reply by pguptaDS** [reply]
Yes thats the direction we'd be going with if we can't delete the assignment right away. Thanks for your help!
